ui/webui/test/helpers/installer.py

```diff
@@ -17,18 +17,40 @@
 
 
 class Installer():
+    _pages = ['installation-language',
+              'installation-destination',
+              'review-configuration',
+              'installation-progress']
+    _start_page = 'installation-language'
+
     def __init__(self, browser):
         self.browser = browser
 
-    def begin_installation(self):
+    def begin_installation(self, expecting_success=True):
+        current_page = self.get_current_page()
         self.browser.click("button:contains('Begin installation')")
+        if expecting_success:
+            self.wait_next_page(current_page)
 
-    def next(self):
+    def next(self, expecting_success=True):
+        current_page = self.get_current_page()
```

Comment:
If you return from get_current_page the page index in the _pages array, you can then remove wait_next_page method.

Instead you can do:

``` py
current_page_idx = self.get_current_page()

self.wait_current_page(current_page_idx + 1)

```